<launch>
    <!-- Pfade zu YAML- und PGM-Dateien als Parameter -->
    <arg name="yaml_file" default="/home/danny/catkin_ws/src/maze_bot_control/maps/maze8.yaml"/>
    <arg name="pgm_file" default="/home/danny/catkin_ws/src/maze_bot_control/maps/maze8.pgm"/>
    
    <arg name="start_x" default="0.03374290466308594"/>
    <arg name="start_y" default="-0.0013123006792739"/>
    
    <arg name="goal_x" default="3.6837387084960938"/>
    <arg name="goal_y" default="-3.395461082458496"/>
    
    <!-- Include der zusÃ¤tzlichen Launch-Datei -->
    <include file="$(find maze_pkg)/launch/maze.launch">
        <arg name="maze_id" value="8"/>
    </include>

    <!-- Map-Loader-Node x -->
    <node pkg="maze_bot_control" type="map_loader.py" name="map_loader" output="screen">
        <param name="yaml_file" value="$(arg yaml_file)"/>
    </node>

    <!-- A*-Planner-Node-->
    <node pkg="maze_bot_control" type="a_star_algorithm.py" name="a_star_planner" output="screen">
        <param name="start_x" value="$(arg start_x)"/>
        <param name="start_y" value="$(arg start_y)"/>
        <param name="goal_x" value="$(arg goal_x)"/>
        <param name="goal_y" value="$(arg goal_y)"/>
    </node>

    <!-- Path-Follower-Node-->
    <node name="controller" pkg="maze_bot_control" type="tb3_driver.py" output="screen">
        <param name="linear_speed" value="0.2" />
        <param name="angular_speed" value="0.5" />
        <param name="distance_threshold" value="0.1" />
        <param name="angle_threshold" value="0.1" />
    </node>

    <!-- Add a static transform between map and odom
    <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_odom"
        args="-0.75 0.75 0 0 0 map odom" />-->
    <!--args="x y z roll pitch yaw frame_id child_frame_id"-->

    <!--<remap from="tf" to="tf_override"/>
    <remap from="tf_static" to="tf_static_override"/>-->
    

</launch>
